### PR TITLE
feat: refractor jupyter

### PR DESCRIPTION
**Pull request by umeshkumhar**
- move jupyter into application
- support private cluster using connectgateway
- fixed IAP bugs

### CODE REVIEW

File: applications/jupyter/workloads.tfvars [hunk 1]
```diff
@@ -15,26 +15,32 @@
 ##common variables  
 ## Need to pull this variables from tf output from previous platform stage
 project_id = "ai-on-gke-jss-sandbox"
+
 ## this is required for terraform to connect to GKE master and deploy workloads
+cluster_name     = "temp-private-cluster"
```
Comment (imreddy13) [inline applications/jupyter/workloads.tfvars:20]: Should we call this something else like "ml-cluster"?

File: applications/jupyter/workloads.tfvars [hunk 2]
```diff
@@ -15,26 +15,32 @@
...
+cluster_location = "us-central1"
+cluster_membership_id = "" # require for private cluster, default: cluster_name
 
 #######################################################
 ####    APPLICATIONS
 #######################################################
 
 ## JupyterHub variables
 namespace = "rag"
```
Comment (imreddy13) [inline applications/jupyter/workloads.tfvars:29]: Maybe something more generic like "ml"

File: applications/jupyter/workloads.tfvars [hunk 3]
```diff
@@ -15,26 +15,32 @@
...
-gcs_bucket = "gcs-bucket"
+gcs_bucket = "rag-demo-bucket"
```
Comment (imreddy13) [inline applications/jupyter/workloads.tfvars:30]: same, prefer something more generic

File: applications/ray/workloads.tfvars [hunk 4]
```diff
@@ -18,15 +18,16 @@
 project_id = "ai-on-gke-jss-sandbox"
 
 ## this is required for terraform to connect to GKE master and deploy workloads
-cluster_name     = "ml-cluster1"
+cluster_name     = "temp-private-cluster"
```
Comment (imreddy13) [inline applications/ray/workloads.tfvars:21]: same, prefer "ml-cluster" and also "ml" for namespace

File: modules/jupyter_iap/charts/iap_jupyter/templates/backend-config.yaml [hunk 5]
```diff
@@ -17,6 +17,14 @@ kind: BackendConfig
 metadata:
   name: {{ .Values.iap.backendConfig.name }}
 spec:
+  healthCheck:
```
Comment (imreddy13) [inline modules/jupyter_iap/charts/iap_jupyter/templates/backend-config.yaml:20]: Why is this needed?

File: applications/jupyter/README.md
Comment (imreddy13) [inline applications/jupyter/README.md:1]: Don't we also need to update the README to reflect the new changes (how to install jupyterhub is different now)?

File: modules/jupyter_iap/variables.tf
Comment (imreddy13) [inline modules/jupyter_iap/variables.tf:1]: Shouldn't there be an additional module under /modules for just installing jupyterhub?

File: applications/jupyter/authentication/README.MD
Comment (imreddy13) [inline applications/jupyter/authentication/README.MD:1]: Shouldn't this whole folder be under /modules/jupyter ?

File: applications/jupyter/main.tf
Comment (imreddy13) [inline applications/jupyter/main.tf:1]: Same, should be under /modules?

File: modules/jupyter_iap/charts/iap_jupyter/templates/backend-config.yaml (hunk 5 above)
Comment (umeshkumhar) [reply modules/jupyter_iap/charts/iap_jupyter/templates/backend-config.yaml:20]: for consistent healthcheck pass and quick healthly server status.

File: applications/jupyter/jupyter_config/config-selfauth.yaml
Comment (imreddy13) [inline applications/jupyter/jupyter_config/config-selfauth.yaml:1]: Should this be under /modules?

File: applications/jupyter/README.md
Comment (umeshkumhar) [reply applications/jupyter/README.md:1]: it will be in different PR by Himanshu

File: modules/jupyter_iap/variables.tf
Comment (umeshkumhar) [reply modules/jupyter_iap/variables.tf:1]: done

File: applications/jupyter/authentication/README.MD
Comment (umeshkumhar) [reply applications/jupyter/authentication/README.MD:1]: done

File: applications/jupyter/jupyter_config/config-selfauth.yaml
Comment (umeshkumhar) [reply applications/jupyter/jupyter_config/config-selfauth.yaml:1]: done

File: applications/jupyter/workloads.tfvars [hunk 6]
```diff
@@ -15,26 +15,32 @@
...
 #######################################################
 ####    APPLICATIONS
 #######################################################
 
 ## JupyterHub variables
-namespace = "rag"
+namespace = "ml"
 gcs_bucket = "gcs-bucket"
 k8s_service_account = "default"
+gcp_service_account ="jupyter-service-account1"
```
Comment (imreddy13) [inline applications/jupyter/workloads.tfvars:32]: can we remove the "1" at the end?
can we make the names consistent between ray/jupyter? like call both vars gcp_service_account and set the values to "ray-system-account" and "jupyter-system-account"?

**Review by imreddy13 — APPROVED** [review]
